community/kernel/src/test/java/org/neo4j/kernel/impl/index/schema/RawBitsTest.java

```diff
@@ -0,0 +1,203 @@
+            Double.NEGATIVE_INFINITY,
+            -Double.MAX_VALUE,
+            Long.MIN_VALUE,
+            Long.MIN_VALUE + 1,
+            Integer.MIN_VALUE,
+            Short.MIN_VALUE,
+            Byte.MIN_VALUE,
+            0,
+            Double.MIN_VALUE,
+            Double.MIN_NORMAL,
+            Float.MIN_VALUE,
+            Float.MIN_NORMAL,
+            1L,
+            1.1d,
+            1.2f,
+            Math.E,
+            Math.PI,
+            (byte) 10,
+            (short) 20,
+            Byte.MAX_VALUE,
+            Short.MAX_VALUE,
+            Integer.MAX_VALUE,
+            33554432,
+            33554432F,
+            33554433,
+            33554433F,
+            33554434,
+            33554434F,
+            9007199254740991L,
+            9007199254740991D,
+            9007199254740992L,
+            9007199254740992D,
+            9007199254740993L,
+            9007199254740993D,
+            9007199254740994L,
+            9007199254740994D,
+            Long.MAX_VALUE,
+            Float.MAX_VALUE,
+            Double.MAX_VALUE,
+            Double.POSITIVE_INFINITY,
+            Double.NaN,
+            Math.nextDown( Math.E ),
+            Math.nextUp( Math.E ),
+            Math.nextDown( Math.PI ),
+            Math.nextUp( Math.PI )
+    );
+
+    @Test
+    public void mustSortInSameOrderAsValueComparator() throws Exception
+    {
+        // given
+        List<Value> values = asValueObjects( objects );
+        List<SchemaNumberKey> schemaNumberKeys = asSchemaNumberKeys( values );
+        Collections.shuffle( values );
+        Collections.shuffle( schemaNumberKeys );
+
+        // when
+        values.sort( Values.COMPARATOR );
+        schemaNumberKeys.sort( layout );
+        List<Value> actual = asValues( schemaNumberKeys );
+
+        // then
+        assertSameOrder( actual, values );
+    }
+
+    @Test
+    public void shouldCompareAllValuesToAllOtherValuesLikeValueComparator() throws Exception
+    {
+        // given
+        List<Value> values = asValueObjects( objects );
+        List<SchemaNumberKey> schemaNumberKeys = asSchemaNumberKeys( values );
+        values.sort( Values.COMPARATOR );
+
+        // when
+        for ( SchemaNumberKey numberKey : schemaNumberKeys )
+        {
+            List<SchemaNumberKey> withoutThisOne = new ArrayList<>( schemaNumberKeys );
+            assertTrue( withoutThisOne.remove( numberKey ) );
+            withoutThisOne = unmodifiableList( withoutThisOne );
+            for ( int i = 0; i < withoutThisOne.size(); i++ )
+            {
+                List<SchemaNumberKey> withThisOneInWrongPlace = new ArrayList<>( withoutThisOne );
+                withThisOneInWrongPlace.add( i, numberKey );
+                withThisOneInWrongPlace.sort( layout );
+                List<Value> actual = asValues( withThisOneInWrongPlace );
+
+                // then
+                assertSameOrder( actual, values );
+            }
+        }
+    }
```

Comment:
I was more thinking of a nested loop where you check that for all combinations `layout.compare` return the same as `Values.COMPARATOR.compare`.